File: day3/Quiz1/solution.py
def getListDiff(lst):
    return len(range(min(lst), max(lst))) + 1 - len(lst) if lst else 0

if __name__ == "__main__":

    print(getListDiff([-10,2,3])) # -10 -9 -8 -7 -6 -5 -4 -3 -2 -1 0 1 2 3

'''
[1, 2, 3, 4] => 0
[] => 0
[1] => 0
[6, 3, 1] => 3
[4, 8, 6] => 2
[1, 11, 8, 12] => 8
'''

File: day3/Quiz1/solution.java
import java.util.List;

public class solution {

    public static void main(String[] args) {
        List<Integer> lst = List.of(4,8,6);
        System.out.println(getListDiff(lst));
        
    }
       
    public static int getListDiff(List<Integer> lst) {

        int result = 0;

        if(lst.size() > 0){
            int max = lst.stream().max(Integer::compare).orElse(0);
            int min = lst.stream().min(Integer::compare).orElse(0);
    
            int oriCnt = lst.size();
            int newCnt = max - min + 1;

            result = newCnt - oriCnt;
        }
        return result;
    }


} 
